File: town_judge.cpp
// Time Complexity : O(n)
// Space Complexity :  O(n)
// Did this code successfully run on Leetcode : Almost. Failing just 5/92 tests
// Any problem you faced while coding this - Not sure what seems to be the issue

class Solution {
public:
    int findJudge(int n, vector<vector<int>>& trust) {
        
        set<int> tr_set;
        int i;
        
        for(i = 0; i < trust.size(); i++)
        {
            tr_set.insert(trust[i][0]);
        }
        
        if(tr_set.size() == n)
        {
            return -1;
        }
        
        for(i = 0; i < n; i++)
        {
            if(tr_set.find(i+1) == tr_set.end())
            {
                return i+1;
            }
        }
        
        return -1;
    }
};

File: TownJudge.swift
// Time Complexity : O(n)
// Space Complexity : O(n)
// Did this code successfully run on Leetcode : yes
// Any problem you faced while coding this : no


// Your code here along with comments explaining your approach
//Find indegree and outdegree. The person who is having the indegree equal to N-1(except himself) then he is the judge.


class Solution {
    func findJudge(_ N: Int, _ trust: [[Int]]) -> Int {
        var indegree = [Int](repeating: 0, count: N)
        
        for person in trust {
            indegree[person[0]-1] -= 1
            indegree[person[1]-1] += 1
        }        
        for (index,person) in indegree.enumerated() {
            if person == N-1 {
                return index+1
            }
        }
        
        return -1
    }
}